diesel_codegen/src/schema_inference/pg.rs

```diff
@@ -38,11 +40,11 @@ table! {
 }
 
 pub fn determine_column_type(cx: &mut ExtCtxt, attr: &ColumnInformation) -> P<ast::Ty> {
-    let tpe = if attr.type_name.starts_with("_") {
-        let subtype = str_to_ident(&capitalize(&attr.type_name[1..]));
+    let tpe = if attr.type_name.ends_with("[]") {
```

Comment:
Informations relating to primary keys living in pg_index, informations about datatypes in pg_types and column related stuff in pg_attributes.
I couldn't get more than one join working, so I decided to rewrite the data type part of the query to use a cast, instead a join to get name of the data type. Unfortunately this results in this changes.